### PR TITLE
Mhd/data source and transformation updates

### PR DESCRIPTION
**Pull request by mhdostal**
Adds a "UserDirections" view and a "Calibration" view along with some LocationDataSource-usage modifications.

### CODE REVIEW

**Comment by mhdostal** [comment]
@philium I've added a couple of user interface elements: a user directions view, which is a simple UILabel in a UIVisualEffects view.  The second one is a view which overlays the SceneView which allows users to "calibrate" the user's camera location via single-finger panning in the SceneView and elevation and heading via two sliders.

In order to get the vertical slider I had to add a transform to the elevation UISlider.  I had difficulty setting the constraints, so I ended up just hardcoding the width of the control (which when rotated is actually the height).  If you can think of a better way to do that, that would be cool.

File: Toolkit/ArcGISToolkit/AR/ArcGISARView.swift [hunk 1]
```diff
@@ -73,29 +73,20 @@ public class ArcGISARView: UIView {
         }() {
         didSet {
             // If we're already tracking, reset tracking to use the new configuration.
-            if isTracking {
-                resetTracking()
+            if isTracking, isUsingARKit {
+                arSCNView.session.run(arConfiguration, options: [.resetTracking])
```
Comment (philium) [inline Toolkit/ArcGISToolkit/AR/ArcGISARView.swift]:
```suggestion
                arSCNView.session.run(arConfiguration, options: .resetTracking)
```

FYI: The type of `options` is `ARSession.RunOptions` and the type of the static member `resetTracking` is `ARSession.RunOptions`. Therefore when there is only one option, you can pass the option directly. By enclosing it in an array literal, it means that a new instance of `ARSession.RunOptions` will be created from an array containing a single instance of `ARSession.RunOptions` (`resetTracking`). So it is more efficient to just pass the static member.

File: Toolkit/ArcGISToolkit/AR/ArcGISARView.swift [hunk 2]
```diff
@@ -448,18 +466,40 @@ extension ArcGISARView: AGSLocationChangeHandlerDelegate {
     
     public func locationDataSource(_ locationDataSource: AGSLocationDataSource, locationDidChange location: AGSLocation) {
         // Location changed.
-        guard let locationPoint = location.position else { return }
+        guard var locationPoint = location.position else { return }
+        
+        print("New Location")
```
Comment (philium) [inline Toolkit/ArcGISToolkit/AR/ArcGISARView.swift]: Did you mean to leave this print statement here?

File: Examples/ArcGISToolkitExamples/ARExample.swift [hunk 3]
```diff
@@ -242,45 +317,149 @@ extension ARExample: ARSCNViewDelegate {
...
+    /// Update the displayed message in the user directions view for the current frame and tracking state.
+    ///
+    /// - Parameters:
+    ///   - frame: The current ARKit frame.
+    ///   - trackingState: The current ARKit tracking state.
+    private func updateUserDirections(_ frame: ARFrame, trackingState: ARCamera.TrackingState) {
+        var message = ""
+        
+        switch trackingState {
+        case .normal where frame.anchors.isEmpty:
```
[the remark this answers is not in the document]
Comment (mhdostal) [reply Examples/ArcGISToolkitExamples/ARExample.swift]: Yes, that's awesome, thanks.

File: Examples/ArcGISToolkitExamples/ARExample.swift [hunk 4]
```diff
@@ -242,45 +317,149 @@ extension ARExample: ARSCNViewDelegate {
...
+    }
+}
+
+// MARK: Calibration View
+extension ARExample {
+    
+    /// Add the calibration view to the view and setup constraints.
+    func addCalibrationView() {
+        calibrationView = CalibrationView(sceneView: arView.sceneView, cameraController: arView.cameraController)
+        guard let calibrationView = calibrationView else { return }
```
[the remark this answers is not in the document]
Comment (mhdostal) [reply Examples/ArcGISToolkitExamples/ARExample.swift]: I refactored this to fix the slider issue.

File: Examples/ArcGISToolkitExamples/ARExample.swift [hunk 5]
```diff
@@ -148,41 +139,123 @@ class ARExample: UIViewController {
...
     @objc func showStatus(_ sender: UIBarButtonItem){
         UIView.animate(withDuration: 0.25) { [weak self] in
             self?.statusViewController?.view.alpha = self?.statusViewController?.view.alpha == 1.0 ? 0.0 : 1.0
         }
     }
+    
+    /// Sets up the toolbar and add it to the view.
+    private func addToolbar() {
+        // Add it to the arView and set up constraints.
+        arView.addSubview(toolbar)
```
[the remark this answers is not in the document]
Comment (mhdostal) [reply Examples/ArcGISToolkitExamples/ARExample.swift]: I've added it the the VC.view; I'm still setting the bottom anchor constraint to `arView.sceneView.attributionTopAnchor` and it seems to work fine.

**Comment by mhdostal** [comment]
@philium - I made the changes you requested, along with some enhancements and general bug fixes.

I made the elevation slider horizontal because I couldn't get it to work vertically.  Interestingly, I also had problems with the height of the view when I did that; to fix it, I'm hiding the attribution bar before showing calibration (`sceneView.isAttributionTextVisible`) which fixed the problems.

Thanks for all your help on that!

**Comment by mhdostal** [comment]
@philium I made the change we talked about for the sliders/labels going under the notch and the touchUpOutside not resetting the slider.

File: Toolkit/ArcGISToolkit/AR/ArcGISARView.swift [hunk 6]
```diff
@@ -289,13 +305,16 @@ public class ArcGISARView: UIView {
         guard let worldTransform = results.first?.worldTransform else { return nil }
         
         // Create our hit test matrix based on the worldTransform location.
+        // right now we ignore the orientation of the plane that was hit to find the point
+        // since we only use horizontal planes, when we will start using vertical planes
+        // we should stop suppressing the quternion rotation to a null rotation (0,0,0,1)
```
Comment (pgruenler) [inline Toolkit/ArcGISToolkit/AR/ArcGISARView.swift]: quternion -> quaternion, missing the 'a'.

**Comment by mhdostal** [comment]
@philium I just added a `Cancel` action to the "Change Scene" action sheet.  Can you take a quick peek (it's two lines of code).  Thanks.

**Pull request merged**